src/machine/machine-dbus.c

```diff
@@ -1003,12 +1001,8 @@ int bus_machine_method_bind_mount(sd_bus_message *message, void *userdata, sd_bu
                 }
 
                 if (make_file_or_directory) {
-                        if (S_ISDIR(st.st_mode))
-                                (void) mkdir_p(dest, 0755);
-                        else {
-                                (void) mkdir_parents(dest, 0755);
-                                (void) mknod(dest, S_IFREG|0600, 0);
-                        }
+                        (void) mkdir_parents(dest, 0755);
+                        (void) make_mount_point_inode_from_stat(&st, dest, 0700);
```

Comment:
@bluca @poettering There is a change in behavior here: now the latest directory is created with mode `0700`, but was previously created with mode `0755`. Is it voluntary?

Reply:
Yes, I think this is what we want. It's the mount point, so the DAC bits after the mount will be those of the source. We want to limit the possibilities of others messing with the mount points before the mount operation.

Did you encounter any issue due to this?

Reply:
No, I was just reading recent commits... and was wondering if this is a mistake or voluntary. Sorry for the noise :-)

Reply:
No worries, always good to ask and double check